File: hospital/src/main/java/com/hospital/service/PrescriptionService.java
package com.hospital.service;

import com.hospital.model.Prescription;

import java.util.List;
import java.util.Optional;

public interface PrescriptionService {
  List<Prescription> getAllPrescriptions();
  Prescription createPrescription(Prescription prescription);
  Optional<Prescription> getPrescriptionById(Long id);
  Prescription updatePrescription(Long id, Prescription prescriptionDetails) throws Exception;
  void deletePrescription(Long id) throws Exception;
}

File: src/app/prescription.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Prescription } from './prescription';

@Injectable({
  providedIn: 'root',
})
export class PrescriptionService {
  private baseUrl = 'http://localhost:8081/hospital/api/v1/prescriptions';

  constructor(private httpClient: HttpClient) {}

  getPrescriptionslist(): Observable<Prescription[]> {
    return this.httpClient.get<Prescription[]>(`${this.baseUrl}`);
  }

  createPrescription(Prescription: Prescription): Observable<Prescription> {
    return this.httpClient.post<Prescription>(`${this.baseUrl}`, Prescription);
  }

  getPrescriptionById(id: number): Observable<Prescription> {
    return this.httpClient.get<Prescription>(`${this.baseUrl}/${id}`);
  }

  updatePrescription(
    id: number,
    Prescription: Prescription
  ): Observable<Object> {
    return this.httpClient.put(`${this.baseUrl}/${id}`, Prescription);
  }

  deletePrescription(id: number): Observable<Object> {
    return this.httpClient.delete(`${this.baseUrl}/${id}`);
  }
}
